client/src/partials/patients/record/patient_record.js

```diff
@@ -1,26 +1,59 @@
 angular.module('bhima.controllers')
 .controller('PatientRecordController', PatientRecordController);
 
-PatientRecordController.$inject = ['$stateParams', 'PatientService', 'NotifyService', 'moment'];
+PatientRecordController.$inject = ['$stateParams', 'PatientService', 'NotifyService', 'moment', 'Upload', '$timeout', '$translate'];
 
-function PatientRecordController($stateParams, Patients, Notify, moment) {
+function PatientRecordController($stateParams, Patients, Notify, moment, Upload, $timeout, $translate) {
   var vm = this;
   var patientID = $stateParams.patientID;
   
   vm.loading = true;
+  vm.uploadFiles = uploadFiles;
+  vm.textButton = $translate.instant('PATIENT_RECORDS.UPLOAD_PICTURE');
+ 
+  refreshPatientsRead();
 
-  /** @fixme if no uuid is provided this will download all the patients through the base url '/' */
-  Patients.read(patientID)
-    .then(function (result) {
-      vm.patient = result;
-      vm.loading = false;
+  function uploadFiles(file, errFiles) {
+    vm.f = file; 
+    if (file) {
+      var imageCheck = file.type.search("image/");
+      if(imageCheck !== -1){
+        file.upload = Upload.upload({
+          url: '/patients/' + patientID + '/pictures', 
+          data: {pictures: file}
+        });
 
-      /** @todo move to service or mysql query */
-      vm.patient.name = [vm.patient.first_name, vm.patient.middle_name, vm.patient.last_name].join(' ');
-      vm.patient.age = moment().diff(vm.patient.dob, 'years');
-    })
-    .catch(function (error) {
-      vm.loading = false;
-      Notify.handleError(error);
-    });
-}
+        file.upload.then(function (response) {
+          Notify.success('FORM.INFOS.PATIENT_SUCC_TRANSFERRED');
+          $timeout(function () {
+            file.result = response.data;
+            refreshPatientsRead();
+          });
+        });
+      } else {
+        Notify.danger('FORM.INFOS.UPLOAD_PICTURE_FAILED');
+      }
+    }   
+  }
+
+  function refreshPatientsRead() {
+    /** @fixme if no uuid is provided this will download all the patients through the base url '/' */
+    Patients.read(patientID)
+      .then(function (result) {
+        vm.patient = result;
+        vm.loading = false;
+
+        if(vm.patient.picture_patient){
+          vm.textButton = $translate.instant('PATIENT_RECORDS.UPDATE_PICTURE');
```

Comment:
Instead of using `$translate()` here, I propose that you do something like this:
```js
vm.uploadButtonText = 'PATIENT_RECORD.UPDATE_PICTURE';

// in the HTML, you can do:
// <button class="btn btn-default" ...>{{ vm.uploadButtonText | translate }}</button>
```
This way, the view does all the presentation, while the controller only switches the logic of the text.